### PR TITLE
`Table` documentation

### PR DESCRIPTION
**Pull request by shaneeza**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/table/README.md
Comment (bruugey) [inline packages/table/README.md:1]: You can delete the contents of this file and replace with DOCUMENTATION.md - I know that it breaks our pattern with read mes but I don't think that we need multiple markdown files and you've accomplished what's necessary in the other file (so just move the contents here)

File: packages/table/OLD_README.md
Comment (bruugey) [inline packages/table/OLD_README.md:1]: Let's rename this v11ReadMe or something similar

Comment (bruugey) [reply packages/table/OLD_README.md:1]: we now have two of these files, we can rm one

**Review by bruugey — APPROVED** [review]
Once you remove the dup'd file, this looks good to me.
Again, we can add examples slowly over time but I do not want to to block on this.

**Comment by shaneeza** [comment]
> Once you remove the dup'd file, this looks good to me. Again, we can add examples slowly over time but I do not want to to block on this.

Once the migration guide is done, I'll merge this in

**Review by bruugey — APPROVED** [review]

**Pull request merged**
